ruff.toml

```diff
@@ -0,0 +1,76 @@
+# Enable rules.
+select = [
+    "A",
+    "C4",
+    "C90",
+    "COM",
+    #"D", # pydocstyle - not enabled temporarily
+    "E",
+    "ERA",
+    "FA",
+    "FIX",
+    "F",
+    "I",
+    "ICN",
+    "ISC",
+    "PLE",
+    "PLC",
+    "PLR",
+    "PLW",
+    "PL",
+    "PYI",
+    "Q",
+    "PTH",
+    "TD",
+    "W"
+]
+
+# Ignore specific rules
+ignore = [
+    "W291", # https://beta.ruff.rs/docs/rules/trailing-whitespace/
+    "PLR0913", # https://beta.ruff.rs/docs/rules/too-many-arguments/
+    "PLR2004", #https://beta.ruff.rs/docs/rules/magic-value-comparison/
+    "PLW0603", #https://beta.ruff.rs/docs/rules/global-statement/
+    "COM812", # Trailing comma missing - disabled temporarily
+    "PLC1901", # Compare-to-empty-string - disabled temporarily
+    "ERA001", # Found commented-out code - disabled temporarily
+    "PLW", # Warning category - disabled temporarily
+    "PLR", # Refactoring category - disabled temporarily
+    "PLC", # Convention category - disabled temporarily
+    "PLE", # Error category - disabled temporarily
+    "ISC", # flake8-implicit-str-concat - disabled temporarily
+    "E501", # Line too long - disabled temporarily
+    "I001", # isort - disabled temporarily
+    ]
+
+# Exclude files and directories
+exclude = ["docs", ".eggs", "setup.py", "example", ".aws-sam", ".git", "dist", ".md", ".yaml", "example/samconfig.toml", ".txt", ".ini"]
+
+# Maximum line length
+line-length = 120
+
+
+#fix = true
+#fixable = ["I", "COM812", "W"]
+
+# See: https://github.com/astral-sh/ruff/issues/128
+typing-modules = ["aws_lambda_powertools.utilities.parser.types"]
+
+[mccabe]
+# Maximum cyclomatic complexity
+max-complexity = 15
+
+[pylint]
+# Maximum number of nested blocks
+max-branches = 15
+# Maximum number of if statements in a function
+max-statements = 70
+
+[isort]
+split-on-trailing-comma = true
```

Comment:
We may need to check other configurations here: https://beta.ruff.rs/docs/settings/#isort

Reply:
for `isort` we should use the profile option so it doesn't conflict with black later on imports.

`profile = "black"`

https://beta.ruff.rs/docs/faq/#how-does-ruffs-import-sorting-compare-to-isort